junit5/src/main/java/com/linecorp/armeria/testing/junit5/client/HttpDataAssert.java

```diff
@@ -0,0 +1,217 @@
+        assertTrue(bytesContains(actual().array(), toByteArray(expected)));
+        return response();
+    }
+
+    /**
+     * Asserts that the actual byte array of {@link HttpData} contains only the given values and nothing else,
+     * in order.
+     * The {@code expected} cannot be null.
+     */
+    public TestHttpResponse arrayContainsExactly(byte... expected) {
+        requireNonNull(expected, "expected");
+        assertTrue(bytesContainsExactly(actual().array(), expected));
+        return response();
+    }
+
+    /**
+     * Asserts that the actual byte array of {@link HttpData} contains only the given values and nothing else,
+     * in order.
+     * The {@code expected} cannot be null.
+     */
+    public TestHttpResponse arrayContainsExactly(Byte[] expected) {
+        requireNonNull(expected, "expected");
+        assertTrue(bytesContainsExactly(actual().array(), toPrimitiveByteArray(expected)));
+        return response();
+    }
+
+    /**
+     * Asserts that the actual byte array of {@link HttpData} contains only the given values and nothing else,
+     * in order.
+     * The {@code expected} cannot be null.
+     */
+    public TestHttpResponse arrayContainsExactly(int... expected) {
+        requireNonNull(expected, "expected");
+        assertTrue(bytesContainsExactly(actual().array(), toByteArray(expected)));
+        return response();
+    }
+
+    /**
+     * Asserts that the length of actual {@link HttpData} is equal to the given one.
+     */
+    public TestHttpResponse lengthIsEqualTo(int expected) {
+        assertEquals(expected, actual().length());
+        return response();
+    }
+
+    /**
+     * Asserts that the {@link String} representation of actual {@link HttpData} is equal to the given one.
+     */
+    public TestHttpResponse stringIsEqualTo(Charset charset, String expected) {
+        requireNonNull(charset, "charset");
+        requireNonNull(expected, "expected");
+        assertEquals(expected, actual().toString(charset));
+        return response();
+    }
+
+    /**
+     * Asserts that the UTF-8 {@link String} representation of actual {@link HttpData} is equal to the given
+     * one.
+     */
+    public TestHttpResponse stringUtf8IsEqualTo(String expected) {
+        requireNonNull(expected, "expected");
+        assertEquals(expected, actual().toStringUtf8());
+        return response();
+    }
+
+    /**
+     * Asserts that the ascii {@link String} representation of actual {@link HttpData} is equal to the given
+     * one.
+     */
+    public TestHttpResponse stringAsciiIsEqualTo(String expected) {
+        requireNonNull(expected, "expected");
+        assertEquals(expected, actual().toStringAscii());
+        return response();
+    }
+
+    /**
+     * Asserts that the {@link String} representation of actual {@link HttpData} contains the given value.
+     */
+    public TestHttpResponse stringContains(Charset charset, String expected) {
```

Comment:
Do you mind also formatting the other methods so that the verb comes first? (following how junit, assertj are naming assertions)

```suggestion
    public TestHttpResponse containsString(Charset charset, String expected) {
```

Reply:
Thanks for the suggestion! When I named the `stringContains` method, my intention was not to convey "the HTTP content contains the parameter string" but rather "when representing HTTP content as a string, it contains the provided string as an argument." If I were to name it more explicitly, it might be something like `toStringAndItWillContain`. The background behind choosing such names was due to the ability to represent HTTP content in various ways, such as UTF-8, ASCII, and others. Initially, I thought of `stringUtf8Contains`, and then I came up with `stringContains`.

As you mentioned, it does seem awkward. How about remove the `stringUtf8Contains` and `stringAsciiContains` methods and simply use shorter method names`contains(String)` and `contains(Charset charset, String expected)`?

Reply:
> my intention was not to convey "the HTTP content contains the parameter string" but rather "when representing HTTP content as a string, it contains the provided string as an argument."

I see, ideally I think we would as a `StringAssert` class then and chain like the following:

```java
assertContent().asString().isEqualTo(...);
```

I think this can be done separately though - what do you think of focusing on byte equality since `HttpData` is essentially a byte array?

Users probably still find this useful - i.e.
```java
assertContent().isEqualTo("".getBytes());
```